### ISSUE TITLE
VNSI Plugin causes periodic channel switching and screws up driver

### ISSUE THREAD

**Issue by Snafu** [FernetMenta/vdr-plugin-vnsiserver#22]
Hi!
I'm running VDR 2.2.0 compiled from source with VNSI Server Plugin (commit 9529e6dc37e775ec355f645714414d10edac3284)  on Ubuntu 14.04 and have a problem that VDR periodically switches to the channel it is currently on with a period of about 11 seconds and that inevitably the mantis driver is screwed and I cant get any picture any more, EPG workds though.
Restarting VDR is not enough, I have to unload and reoad the mantis Kernel module.

Weird thing is I can't really find any errors in the log except for the constant channel switching.
Any ideas? If I run the streamdev plugin instead everything is fine.

05:00.0 Multimedia controller: Twinhan Technology Co. Ltd Mantis DTV PCI Bridge Controller [Ver 1.0] (rev 01)
        Subsystem: TERRATEC Electronic GmbH Device 1178
        Flags: bus master, medium devsel, latency 32, IRQ 16
        Memory at f0101000 (32-bit, prefetchable) [size=4K]
        Kernel driver in use: Mantis

05:01.0 Multimedia controller: Twinhan Technology Co. Ltd Mantis DTV PCI Bridge Controller [Ver 1.0] (rev 01)
        Subsystem: TERRATEC Electronic GmbH Device 1178
        Flags: bus master, medium devsel, latency 32, IRQ 17
        Memory at f0100000 (32-bit, prefetchable) [size=4K]
        Kernel driver in use: Mantis

**Comment by FernetMenta** [comment]
you can switch off epg scan while streaming.

**Comment by Snafu** [comment]
No EPG is the only thing that is still working then. I guess it takes data from the epg.dat file and checks with the system time how far it is within the show.
Something is screwing up the driver and I guess it's the constant channel switching.

**Comment by FernetMenta** [comment]
i said you can switch off epg scan while streaming. why do you answer no? I don't get the sense of that.

**Comment by Snafu** [comment]
Yes, but it has nothing to do with the EPG scan feature. Same behaviour regardless of having the setting on or off. As soon as the vnsi-server plugin is loaded it periodically switches to the active channel with a period of 11 seconds. It does this as soon as VDR is started and not only once a client is connected.

**Comment by Snafu** [comment]
Yes, but it has nothing to do with the EPG scan feature. Same behaviour regardless of having the setting on or off. As soon as the vnsi-server plugin is loaded it periodically switches to the active channel with a period of 11 seconds. It does this as soon as VDR is started and not only once a client is connected.

**Comment by FernetMenta** [comment]
please define "active channel" when no client is connected. if no client is connected, vnsi does not play any channel. so what is the active channel? how do you observe vnsi doing something? please elaborate on this.

**Comment by Snafu** [comment]
Ok, I did a little reaserch and testing and while the problem with the driver turns out to be due to a faulty PCIe to PCI chipset [1], the constant channel switching only occurs with the -d option to set it as the primary device. I don't know if this has any negative consequences besides filling up the log (and speeding up the time until the driver crashes in my case).

[1] https://bugzilla.kernel.org/show_bug.cgi?id=38632

**Comment by FernetMenta** [comment]
> he constant channel switching only occurs with the -d option

thanks for this info. I will investigate

**Comment by FernetMenta** [comment]
https://github.com/FernetMenta/vdr-plugin-vnsiserver/pull/27 should fix the unwanted channel switches
